token_update_nfts_transaction_unit_test.go

```diff
@@ -0,0 +1,221 @@
+//go:build all || unit
+// +build all unit
+
+package hedera
+
+import (
+	"testing"
+
+	"github.com/hashgraph/hedera-protobufs-go/services"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	protobuf "google.golang.org/protobuf/proto"
+)
+
+func TestUnitTokenUpdateNftsTransactionValidate(t *testing.T) {
+	t.Parallel()
+
+	client, err := _NewMockClient()
+	client.SetLedgerID(*NewLedgerIDTestnet())
+	require.NoError(t, err)
+	client.SetAutoValidateChecksums(true)
+	require.NoError(t, err)
+	tokenID, err := TokenIDFromString("0.0.123-esxsf")
+	require.NoError(t, err)
+
+	tokenUpdate := NewTokenUpdateNfts().
+		SetTokenID(tokenID).
+		SetSerialNumbers([]int64{1, 2, 3}).
+		SetMetadata([]byte("metadata"))
+
+	err = tokenUpdate.validateNetworkOnIDs(client)
+	require.NoError(t, err)
+}
+
+func TestUnitTokenUpdateNftsTransactionValidateWrong(t *testing.T) {
+	t.Parallel()
+
+	client, err := _NewMockClient()
+	client.SetLedgerID(*NewLedgerIDTestnet())
+	require.NoError(t, err)
+	client.SetAutoValidateChecksums(true)
+	require.NoError(t, err)
+	tokenID, err := TokenIDFromString("0.0.123-rmkykd")
+	require.NoError(t, err)
+
+	tokenUpdate := NewTokenUpdateNfts().
+		SetTokenID(tokenID).
+		SetSerialNumbers([]int64{1, 2, 3}).
+		SetMetadata([]byte("metadata"))
+
+	err = tokenUpdate.validateNetworkOnIDs(client)
+	assert.Error(t, err)
+	if err != nil {
+		assert.Equal(t, "network mismatch or wrong checksum given, given checksum: rmkykd, correct checksum esxsf, network: testnet", err.Error())
+	}
+}
+
+func TestUnitTokenUpdateNftsTransactionGet(t *testing.T) {
+	t.Parallel()
+
+	checksum := "dmqui"
+	tokenID := TokenID{Token: 3, checksum: &checksum}
+
+	newKey, err := PrivateKeyGenerateEd25519()
+	require.NoError(t, err)
+
+	nodeAccountID := []AccountID{{Account: 10}}
+	transactionID := TransactionIDGenerate(AccountID{Account: 324})
+
+	client, err := _NewMockClient()
+	client.SetLedgerID(*NewLedgerIDTestnet())
+	require.NoError(t, err)
+	client.SetAutoValidateChecksums(true)
+
+	transaction, err := NewTokenUpdateNfts().
+		SetTokenID(tokenID).
+		SetSerialNumbers([]int64{1, 2, 3}).
+		SetTransactionID(transactionID).
+		SetNodeAccountIDs(nodeAccountID).
+		SetMetadata([]byte("metadata")).Freeze()
+
+	require.NoError(t, err)
+	transaction.GetTokenID()
+	transaction.GetSerialNumbers()
```

Comment:
Not sure if calling these getters actually do something, but saw that from other unit tests

Reply:
I guess that at least we would have to check the values after the get otherwise looks like we will not be needing these.

Reply:
Added `require.NotNil` or `require.Nil`to check to values